src/pages/blog/http-and-lambda.mdx

```diff
@@ -0,0 +1,338 @@
+
+(defn -handleRequest [_ is os context]
+  (let [input (slurp is)]
+    (println "input:" input)
+    ; If the word "error" is found anywhere in the input
+    (when (re-find #"error" input)
+      (throw (ex-info "test" {:my "error"})))
+    ; If the word "invalid" is found anywhere in the input
+    (if (re-find #"invalid" input)
+      (spit os "{")
+      (spit os "{}"))))
+```
+
+(See my [last post](https://juxt.pro/blog/plain-clojure-lambda) for how I set
+the project up)
+
+Please note that the input is **not** parsed, so any errors that come parsing
+the input come from AWS. Also note that the output does not conform to any of
+the formats required, this tests how each option behaves in this case.
+
+Most services have multiple ways of configuring them, the methods I'm covering
+are:
+- [Plain Function URLs](#plain-function-urls)
+- [Function URLs + CloudFront](#function-urls--cloudfront)
+- [ALB](#alb)
+- [API Gateway w/ HTTP API](#api-gateway-w-http-api)
+- [API Gateway w/ REST API](#api-gateway-w-rest-api)
+- [API Gateway w/ REST API & Lambda Proxy Integration](#api-gateway-w-rest-api--lambda-proxy-integration)
+
+
+
+## Which should you choose?
+
+Before we start, this article doesn't aim to judge which option of these fits
+your requirements. But if you want to know which to check out first I would
+suggest:
+- If you don't want a custom domain: [Plain Function URLs](#plain-function-urls)
+- If you *do* want a custom domain: [Function URLs + CloudFront](#function-urls--cloudfront)
+- For everything else I'd suggest: [ALBs](#alb) or [API Gateway w/ REST & Lambda Proxy Integration](#api-gateway-w-rest-api--lambda-proxy-integration)
+
+
+
+## Plain Function URLs
+
+This is by far the easiest option to setup: [just enable it and set two
+settings](https://docs.aws.amazon.com/lambda/latest/dg/urls-configuration.html#create-url-console)
+and you're done!
+
+The (small) downside is that you get a randomly generated url of the form:
+`https://<url-id>.lambda-url.<region>.on.aws`, but for many use cases this is
+just fine.
+
+As with most of these options, the [request](https://docs.aws.amazon.com/lambda/latest/dg/urls-invocation.html#urls-request-payload)
+and [response](https://docs.aws.amazon.com/lambda/latest/dg/urls-invocation.html#urls-response-payload)
+formats are in JSON. It is worth noting is if the response **does not** match
+this format then a `200` response is assumed.
+
+Here are the results of my tests:
+
+| Test                  | Status Code   | Response Body           | `input` in lambda          |
+| ---                   | ---           | ---                     | ---                        |
+| Valid JSON request    | `200`         | `{}`                    | The JSON formatted request |
+| Invalid JSON request  | `200`         | `{}`                    | The JSON formatted request |
+| Invalid JSON response | `200`         | `{`                     | The JSON formatted request |
```

Comment:
is it worth calling these out as surprising? (were they surprising to you?)

Reply:
Note to self: Maybe it's worth talking about what my initial expectations were in general for all these services?